sdk/src/main/scala/com/horizen/account/network/SyncStatusActor.scala

```diff
@@ -0,0 +1,227 @@
+import com.horizen.account.api.rpc.types.SyncStatus
+import com.horizen.account.network.SyncStatusActor.ReceivableMessages.{NotifySyncStart, NotifySyncStop, ReturnSyncStatus}
+import com.horizen.block.{SidechainBlockBase, SidechainBlockHeaderBase}
+import com.horizen.chain.AbstractFeePaymentsInfo
+import com.horizen.params.NetworkParams
+import com.horizen.storage.AbstractHistoryStorage
+import com.horizen.transaction.Transaction
+import sparkz.core.NodeViewHolder.CurrentView
+import sparkz.core.NodeViewHolder.ReceivableMessages.GetDataFromCurrentView
+import sparkz.core.network.NodeViewSynchronizer.ReceivableMessages.SemanticallySuccessfulModifier
+import sparkz.core.transaction.MemoryPool
+import sparkz.util.SparkzLogging
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{Await, ExecutionContext}
+
+class SyncStatusActor[
+    TX <: Transaction, H <: SidechainBlockHeaderBase,
+    PMOD <: SidechainBlockBase[TX, H],
+    SI <: SidechainSyncInfo,
+    FPI <: AbstractFeePaymentsInfo,
+    HSTOR <: AbstractHistoryStorage[PMOD, FPI, HSTOR],
+    HIS <: AbstractHistory[TX, H, PMOD, FPI, HSTOR, HIS],
+    MS <: AbstractState[TX, H, PMOD, MS],
+    VL <: Wallet[SidechainTypes#SCS, SidechainTypes#SCP, TX, PMOD, VL],
+    MP <: MemoryPool[TX, MP]
+  ]
+(
+  settings: SidechainSettings,
+  sidechainNodeViewHolderRef: ActorRef,
+  params: NetworkParams
+)
+  extends Actor with SparkzLogging {
+
+  type View = CurrentView[HIS, MS, VL, MP]
+
+  private var syncStatus: Boolean = _
+  private var highestBlockTimestamp: Long = _
+
+  var startingBlock: Int = _
+  var highestBlock: Int = _
+
+  private val confidenceParameter: Int = 1
+  private val previousBlocksRange: Int = 1
+
+  lazy val timeoutDuration: FiniteDuration = settings.sparkzSettings.restApi.timeout
+  implicit lazy val timeout: Timeout = Timeout(timeoutDuration)
+
+  override def preStart(): Unit = {
+    context.system.eventStream.subscribe(self, classOf[GetHighestBlockTimestamp])
+    context.system.eventStream.subscribe(self, classOf[SemanticallySuccessfulModifier[PMOD]])
+  }
+
+  override def postStop(): Unit = {
+    log.debug("SidechainBlock Actor is stopping...")
+    super.postStop()
+  }
+
+  def retrieveCurrentBlock(sidechainNodeView: View): (PMOD) = {
+    sidechainNodeView.history.getBestBlock
+  }
+
+  def retrieveCurrentHeight(sidechainNodeView: View): (Int) = {
+    sidechainNodeView.history.getCurrentHeight
+  }
+
+  // retrieve the timestamp from the block recently applied to the state
+  // if the difference between the highestBlockTimestamp and the timestamp of the block applied is less than
+  // the block rate multiplied for a confidence parameter we set the sync status to false.
+  // This means that the node is synched with the rest of the network.
+  def processNewBlockApplied(sidechainBlock: PMOD): Unit = {
+    if((highestBlockTimestamp - sidechainBlock.timestamp).compareTo(params.consensusSlotsInEpoch * confidenceParameter) < 0) {
```

Comment:
do you mean `consensusSecondsInSlot` instead of `consensusSlotsInEpoch` ?
`consensusSlotsInEpoch` has no relation to the time

Reply:
yes, `consensusSlotsInEpoch` is the wrong field.
fixed